File: bsba_be/src/main/java/com/be/bsba/entity/Booking.java
package com.be.bsba.entity;

import com.be.bsba.constant.BookingStatus;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.UUID;

@Entity
@Table(name = "bookings")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Booking {

    @Id
    @GeneratedValue
    private UUID id;

    private Integer participantCount;

    private BigDecimal totalPrice;

    @Column(columnDefinition = "TEXT")
    private String note;

    @Enumerated(EnumType.STRING)
    private BookingStatus status;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "store_id")
    private Store store;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "slot_id")
    private StoreTimeSlot slot;

    @Builder.Default
    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private OffsetDateTime createdAt = OffsetDateTime.now();

    @Builder.Default
    @UpdateTimestamp
    @Column(nullable = false)
    private OffsetDateTime updatedAt = OffsetDateTime.now();

    @OneToMany(mappedBy = "booking", cascade = CascadeType.ALL)
    private java.util.List<BookingGame> games;
}

File: bsba_fe/lib/data/models/booking.dart
class Booking {
  const Booking({
    required this.userId,
    required this.storeId,
    required this.slotId,
    required this.participantCount,
    required this.totalPrice,
    this.note,
  });

  final String userId;
  final String storeId;
  final String slotId;
  final int participantCount;
  final int totalPrice;
  final String? note;

  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'storeId': storeId,
      'slotId': slotId,
      'participantCount': participantCount,
      'totalPrice': totalPrice,
      if (note != null && note!.isNotEmpty) 'note': note,
    };
  }
}
